packages/world/test/WorldProxyFactory.t.sol

```diff
@@ -0,0 +1,156 @@
+    assertEq(NamespaceOwner.get(ROOT_NAMESPACE_ID), account1);
+
+    // Deploy a second world
+
+    // User defined bytes for create2
+    // unchecked for the fuzzing test
+    bytes memory _salt2 = abi.encode(salt2);
+
+    // Address we expect for second World
+    calculatedAddress = calculateAddress(
+      worldFactoryAddress,
+      keccak256(abi.encode(account1, 0)),
+      type(World).creationCode
+    );
+
+    // Check for HelloWorld event from World
+    vm.expectEmit(true, true, true, true);
+    emit IWorldEvents.HelloWorld(WORLD_VERSION);
+
+    worldAddress = worldFactory.deployWorld(_salt2);
+
+    worldImplementationAddress = address(uint160(uint256(vm.load(worldAddress, IMPLEMENTATION_SLOT))));
+
+    // Set the store address manually
+    StoreSwitch.setStoreAddress(worldAddress);
+
+    // Confirm correct Core is installed
+    assertTrue(InstalledModules.get(address(initModule), keccak256(new bytes(0))));
+
+    // Confirm the msg.sender is owner of the root namespace of the new world
+    assertEq(NamespaceOwner.get(ROOT_NAMESPACE_ID), account1);
+
+    // Expect revert when deploying world with same bytes salt as already deployed world
+    vm.expectRevert();
+    worldFactory.deployWorld(_salt1);
+
+    // Expect revert when initializing world as not the creator
+    vm.expectRevert(
+      abi.encodeWithSelector(IWorldErrors.World_AccessDenied.selector, ROOT_NAMESPACE_ID.toString(), account1)
+    );
+    IBaseWorld(address(worldAddress)).initialize(initModule);
+
+    // Deploy a new world
+    address newWorldImplementationAddress = address(new World());
+
+    // Expect revert when changing implementation as not root namespace owner
+    vm.startPrank(account2);
+    vm.expectRevert(
+      abi.encodeWithSelector(IWorldErrors.World_AccessDenied.selector, ROOT_NAMESPACE_ID.toString(), account2)
+    );
+    WorldProxy(payable(worldAddress)).setImplementation(newWorldImplementationAddress);
+
+    vm.startPrank(account1);
+
+    // Set proxy implementation to new world
+    startGasReport("set WorldProxy implementation");
+    WorldProxy(payable(worldAddress)).setImplementation(newWorldImplementationAddress);
+    endGasReport();
+
+    worldImplementationAddress = address(uint160(uint256(vm.load(worldAddress, IMPLEMENTATION_SLOT))));
+    assertEq(worldImplementationAddress, newWorldImplementationAddress);
+
+    // Confirm correct Core is installed
+    assertTrue(InstalledModules.get(address(initModule), keccak256(new bytes(0))));
+
+    // Confirm the msg.sender is owner of the root namespace of the new world
+    assertEq(NamespaceOwner.get(ROOT_NAMESPACE_ID), account1);
+
+    // Test we can execute a world function
+    IBaseWorld(address(worldAddress)).registerDelegation(account2, UNLIMITED_DELEGATION, new bytes(0));
+    assertEq(ResourceId.unwrap(UserDelegationControl.get(account1, account2)), ResourceId.unwrap(UNLIMITED_DELEGATION));
```

Comment:
i was a little confused for a sec where `account1` comes from here; as a general best practice we should use `vm.prank` over `vm.startPrank` to make it easier to follow